### PR TITLE
Docs: Add a "seealso" section to the module docs

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
Something I wanted for a longer time was a separate section to list related modules and/or related references. This clears up the notes section for things that are actual notes.

So you can add a section in your module documentation and four types of references are possible.
```yaml
seealso:

# Reference by module name
- module: aci_tenant

# Reference by module name, including description
- module: aci_tenant
  description: ACI module to create tenants on a Cisco ACI fabric.

# Reference by rST documentation anchor
- ref: aci_guide
  description: Detailed information on how to manage your ACI infrastructure using Ansible.

# Reference by Internet resource
- name: APIC Management Information Model reference
  description: Complete reference of the APIC object model.
  link: https://developer.cisco.com/docs/apic-mim-ref/
```

This PR also includes:
- Implements `ansible-doc` support
- Implements schema support for the `seealso` options
- Updates to the development documentation
- Rename filter `convert_symbols_to_format` to `rst_ify`, cfr the existing `html_ify` and `tty_ify` filters
  - This makes the existing template a lot easier to read and fixes the confusion I had myself rereading the template (again).
- We fixed the possible suboption types (which was limited to 'bool' only)

##### ISSUE TYPE
- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME
Module docs

##### ANSIBLE VERSION
v2.8

##### ADDITIONAL INFORMATION
**Before:**
screenshot from 2018-09-21 03-09-14

**After:**
screenshot from 2018-09-21 03-30-55

**ansible-doc** output
```
SEE ALSO:
      * Module aci_bd
           The official documentation on the aci_bd module.
           https://docs.ansible.com/ansible/devel/modules/aci_bd_module.html
      * Module aci_tenant
           The official documentation on the aci_tenant module.
           https://docs.ansible.com/ansible/devel/modules/aci_tenant_module.html
      * Ansible documentation [aci_guide]
           Detailed information on how to manage your ACI infrastructure using Ansible.
           https://docs.ansible.com/ansible/latest/#stq=aci_guide&stp=1
      * Ansible documentation [aci_dev_guide]
           Detailed guide on how to write your own Cisco ACI modules to contribute.
           https://docs.ansible.com/ansible/latest/#stq=aci_dev_guide&stp=1
```

The documentation for this feature looks like this:
screenshot from 2018-09-21 17-24-07

**TODO** (see #45985):
- Currently ref-type references cannot be converted to a title + link, as Sphinx takes care of this internally. The solution is to make a searchable link that brings the user to the document based on the reference name.

### CODE REVIEW

**Comment by dagwieers** [comment]
cc @felixfontein

**Review by gundalow — DISMISSED** [review]
This looks like a good addition. We will review in more detail after AnsibleFest

`schema.py` and `/developing_modules_documenting` will also need updating

Added to https://etherpad.openstack.org/p/ansible-summit-october-2018-documentation

**Comment by dagwieers** [comment]
@gundalow If you look at the code, I do modify *schema.py*, however it does not work and I have no clue how it is supposed to work. Value should be a list of dictionaries, but the test still does not accept it.

And I think that's why ansible-doc is not working here either. Always returns None.

File: test/sanity/validate-modules/schema.py
Comment (gundalow) [inline test/sanity/validate-modules/schema.py]: I wonder if these should be string (singular) rather than plural `*string_types`

Comment (dagwieers) [reply test/sanity/validate-modules/schema.py]: ~Well, there are multiple string types, I don't think we can work around that.~
Maybe I fixed what you were point at :-)

**Comment by dagwieers** [comment]
This PR is now no longer a WIP. Please re-review !

**Comment by gundalow** [comment]
On hold till after AnsibleFest

**Comment by dagwieers** [comment]
@acozine Can we get this moving ?

**Comment by gundalow** [comment]
Once live we should mention this in https://github.com/ansible/community/issues/346

File: lib/ansible/cli/doc.py
Comment (gundalow) [inline lib/ansible/cli/doc.py]: Can we extract the version of ansible in use and use that rather than `/devel/` (or at least default to `/latest/`?

Comment (acozine) [reply lib/ansible/cli/doc.py]: +1 to this - we don't want to redirect users from the 2.5 docs to `devel` without any warning.

Comment (dagwieers) [reply lib/ansible/cli/doc.py]: Is there a variable that contains the version ?

Comment (dagwieers) [reply lib/ansible/cli/doc.py]: I reverted to using `latest`, as I don't see how we can get to the (major) version from ansible-doc at this time. We can improve this later if necessary.

PS I think using `/latest/` is probably better in any case, since we do specify when parameters are introduced and deprecated. And newer docs may have better information and examples.

**Comment by dagwieers** [comment]
Ok, another re-review required. Thanks !

**Comment by gundalow** [comment]
Look at how `ansible-doc --version` works to get the current version

**Comment by dagwieers** [comment]
The problem using the version string is that it has no knowledge about whether e.g. 2.7 is available on the docsite. Or whether it is `latest` or `devel`. During the several stages of release this may or may not be the case. So I think using `latest` is the safest bet overal.

And we should focus on making it easier to jump between versions on the docsite, which IMO has the highest priority for end-user experience. (Knowing most people end up on the wrong/older documentation through Google anyway).

File: lib/ansible/cli/doc.py
Comment (dagwieers) [reply lib/ansible/cli/doc.py]: @acozine I think you misinterpret what is going on. The link is only exposed when using ansible-doc. So you won't be redirected from 2.5 docs to `devel` (now `latest`), because those references will go to the documentation you are using.

These links are only for people who run ansible-doc on the CLI. In that case we have to provide a link, and we cannot (assume) that a versioned-link exists for the command being used. So I think using `latest` is a good trade-off. Currently these references would simply fail to work. There would not be a link whatsoever, and that's what this proposal fixes too.

**Comment by gundalow** [comment]
Dag, good point. Let's stick with `/latest/`

**Comment by dagwieers** [comment]
Ok, so anything else needed before we can merge this ?

**Comment by acozine** [comment]
@dagwieers could the `seealso` section use the existing, documented link syntax to link to URLs and module docs? (`U(https:external_url)` for links and `M(module_name)` for modules)

**Comment by dagwieers** [comment]
@acozine It is a YAML dictionary, not a free text-field. Because we don't let the user do the markup, it is a listing of information we present depending on where it is being used. So it doesn't make sense.

For instance, for the module entry we depend on the Sphinx references to automatically fill in the title.

**Comment by dagwieers** [comment]
@acozine @gundalow I would like to implement `seealso:` now for the windows/python modules that relate to each other (we now use notes) and for the modules that relate to each other (typically copy + template, copy + rsync,  command + shell + raw +script, etc...) and for the ACI modules.

We probably want to communicate this to other authors too.

**Pull request merged**
